runtimeconfig/manager.go

```diff
@@ -0,0 +1,210 @@
+package runtimeconfig
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/grafana/dskit/services"
+)
+
+// Loader loads the configuration from file.
+type Loader func(r io.Reader) (interface{}, error)
+
+// Config holds the config for an Manager instance.
+// It holds config related to loading per-tenant config.
+type Config struct {
+	ReloadPeriod time.Duration `yaml:"period"`
+	// LoadPath contains the path to the runtime config file, requires an
+	// non-empty value
+	LoadPath string `yaml:"file"`
+	Loader   Loader `yaml:"-"`
+}
+
+// RegisterFlags registers flags.
+func (mc *Config) RegisterFlags(f *flag.FlagSet) {
```

Comment:
We could add a prefix here, but it would a be a breaking change for cortex/tempo/loki

Reply:
I would keep it as is for now. We can always add `RegisterFlagsWithPrefix` in the future, if needed.